cub/test/catch2_test_env_launch_helper.h

```diff
@@ -0,0 +1,567 @@
+/******************************************************************************
+ * Copyright (c) 2011-2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the NVIDIA CORPORATION nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
+
+#pragma once
+
+#include <thrust/system/cuda/detail/core/triple_chevron_launch.h>
+
+#include <cuda/std/optional>
+
+#include <c2h/catch2_test_helper.h>
+
+//! @file
+//! This file contains utilities for device-scope API tests of environment APIs.
+//!
+//! Device-scope API in CUB can be launched from the host or as part of cuda graph.
+//! Utilities in this file facilitate testing in both cases.
+//!
+//!
+//! ```
+//! // Add PARAM to make CMake generate a test for both host and device launch:
+//! // %PARAM% TEST_LAUNCH lid 0:1:2
+//!
+//! // Declare CDP wrapper for CUB API. The wrapper will accept the same
+//! // arguments as the CUB API. The wrapper name is provided as the second argument.
+//! DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Sum, cub_reduce_sum);
+//!
+//! C2H_TEST("Reduce test", "[device][reduce]")
+//! {
+//!   // ...
+//!   // Invoke the wrapper from the test. It'll allocate temporary storage and
+//!   // invoke the CUB API on the host or device side while checking return
+//!   // codes and launch errors.
+//!   cub_reduce_sum(d_in, d_out, n, env);
+//! }
+//!
+//! ```
+//!
+//! Consult with `test/catch2_test_cdp_wrapper.cu` for more usage examples.
```

Comment:
Important: `catch2_test_cdp_wrapper.cu` seems to not exist before and after this PR. Is the file missing or the file name wrong here?